.github/CONTRIBUTING.md

```diff
@@ -1,9 +1,89 @@
 # Contributing to Trino
 
-## Contributor License Agreement ("CLA")
 
-In order to accept your pull request, we need you to [submit a CLA](https://github.com/trinodb/cla).
+This is the process we suggest for contributions.  This process is designed to
+reduce the burden on project reviewers, impact on other contributors, and to
+keep the amount of rework from the contributor to a minimum.
 
-## License
+By contributing to Trino, you agree that your contributions will be licensed
+under the [Apache License Version 2.0 (APLv2)](../LICENSE).
 
-By contributing to Trino, you agree that your contributions will be licensed under the [Apache License Version 2.0 (APLv2)](../LICENSE).
+Find more information at [the Trino site](https://trino.io/development/).
+
+## Contributor License Agreement
+
+In order to accept your pull request, we need you to [submit a contributor
+license agreement (CLA)](https://github.com/trinodb/cla).
+
+## Discussion
+
+Start a discussion by creating a Github
+[issue](https://github.com/trinodb/trino/issues), or asking on
+[Slack](https://trino.io/slack.html) (unless the change is trivial).
+
+* This step helps you identify possible collaborators and reviewers.
+* Does the change align with technical vision and project values?
+* Will the change conflict with another change in progress? If so, work with
+  others to minimize impact.
+* Is this change large?  If so, work with others to break into smaller steps.
+
+## Implementation
+
+Implement the change.
+
+* If the change is large, post a preview Github
+  [pull request](https://github.com/trinodb/trino/pulls) with the title
+  prefixed with `[WIP]`, and share with collaborators.
+* Include tests and documentation as necessary.
+
+
+## Pull request
+
+Create a Github [pull request](https://github.com/trinodb/trino/pulls) (PR).
+
+* Make sure the pull request passes the tests in the Continuous Integration
+  (CI) workflows.
+* If known, request a review from an expert in the area changed.  If unknown,
+  ask for help on [Slack](https://trino.io/slack.html).
+
+There are some tests that use external services, like Google BigQuery, and require
+additional credentials. The Trino project cannot share these credentials with
+contributors, so it runs such tests in its CI workflows only on branches in the
+Trino repository, not in contributor forks.
+
+Trino project maintainers (members with write access to the repository) can
+schedule additional workflow runs after reviewing a PR and adding a comment
+like:
+```
+/test-with-secrets sha=<all-40-characters>
```

Comment:
can the comment contain any other lines?
like other words, mentions, etc.?

Reply:
Yes, only the first line must be a command and cannot contain any extra characters.